### ISSUE TITLE
Allow to change camera position and the point it looks at

### ISSUE THREAD

**Issue by qwerty287** [maplibre/maplibre-gl-js#1414]
## Motivation

I'd like to follow a route on a 3D terrain map, which is hard because I need to set the bearing, the map center etc. but also have to check that there isn't a mountain between the camera and the route which hides it.

## Design + User Mock-up

Add something similar to Mapbox' FreeCamera option (https://docs.mapbox.com/mapbox-gl-js/api/properties/#freecameraoptions, example https://docs.mapbox.com/mapbox-gl-js/example/free-camera-path/) which allows to set two values of the camera:
1. the location of the camera (latitude, longitude, altitude)
2. the point the camera looks at

There are two options for me how the API could look like:
- copy the mapbox API
```js
const camera = map.getFreeCameraOptions();

camera.position = maplibregl.MercatorCoordinate.fromLngLat({
    lng: cameraLongitude,
    lat: cameraLatitude
  },
  cameraAltitude
);

camera.lookAtPoint({
  lng: lookAtLongitude,
  lat: lookAtLatitude
});

map.setFreeCameraOptions(camera);
```
- implement it directly on the map object (which makes it a bit more unreadable if you add more options)
```js
map.cameraPosition = maplibregl.MercatorCoordinate.fromLngLat({
    lng: cameraLongitude,
    lat: cameraLatitude
  },
  cameraAltitude
);

map.lookAtPoint({
  lng: lookAtLongitude,
  lat: lookAtLatitude
});
```

Maplibre should check that there isn't a mountain between the camera and the route (if using it with a low pitch) that could hide the route.

**Comment by HarelM** [comment]
Sound very similar to #1123 and #1318 (while different APIs though).
Please let me know if this is a duplicate or not...

**Comment by prozessor13** [comment]
Hallo,

we are already working on this, please be patient.

**Comment by HarelM** [comment]
Closed by #1427
